crates/yas_core/src/tests/test_contracts/test_yas_pool.cairo

```diff
@@ -486,81 +491,1014 @@ mod YASPoolTests {
+                    );
+                    assert(
+                        yas_pool.ticks(IntegerTrait::<i32>::new(0, false)).liquidity_gross == 160,
+                        'wrong liquidity_gross amount 10'
+                    );
+                    assert(
+                        yas_pool.ticks(yas_pool.tick_spacing()).liquidity_gross == 150,
+                        'wrong liquidity_gross amount 11'
+                    );
+                    assert(
+                        yas_pool
+                            .ticks(
+                                IntegerTrait::<i32>::new(tick_spacing(FeeAmount::MEDIUM) * 2, false)
+                            )
+                            .liquidity_gross == 60,
+                        'wrong liquidity_gross amount 12'
+                    );
+                }
+
+
+                // TODO: 'removes liquidity from liquidityGross'
+                //#[test]
+                //#[available_gas(200000000)]
+                fn test_remove_liquidityGross() {
+                    //to implement:
+                    //await mint(wallet.address, -240, 0, 100) -done
+                    //await mint(wallet.address, -240, 0, 40) -done
+                    //await pool.burn(-240, 0, 90)
+                    //expect((await pool.ticks(-240)).liquidityGross).to.eq(50)
+                    //expect((await pool.ticks(0)).liquidityGross).to.eq(50)
+
+                    let (yas_pool, token_0, token_1, mint_callback, min_tick, max_tick) = setup();
+                    mint_callback
+                        .mint(
+                            yas_pool.contract_address,
+                            WALLET(),
+                            IntegerTrait::<i32>::new(240, true),
+                            IntegerTrait::<i32>::new(0, false),
+                            100
+                        );
+                    mint_callback
+                        .mint(
+                            yas_pool.contract_address,
+                            WALLET(),
+                            IntegerTrait::<i32>::new(240, true),
+                            IntegerTrait::<i32>::new(0, false),
+                            40
+                        );
+                    //burn
+                    assert(1 == 2, 'burn() func doesnt exist yet');
+
+                    assert(
+                        yas_pool.ticks(IntegerTrait::<i32>::new(240, true)).liquidity_gross == 500,
+                        'wrong liquidity_gross amount'
+                    );
+                    assert(
+                        yas_pool.ticks(IntegerTrait::<i32>::new(0, false)).liquidity_gross == 50,
+                        'wrong liquidity_gross amount 2'
+                    );
+                }
+
+
+                // TODO: 'clears tick lower if last position is removed'
+                //#[test]
+                //#[available_gas(200000000)]
+                fn test_clear_tick_lower() {
```

Comment:
Please comment this function with a proper message saying that this will be used once the burn function is merged.

Reply:
done